File: 2_C++(VisualStudio)/2_workspace_MFC_Direct2D/1_SideViewGameSample/src/src/PowerUpBlock1Object.cpp
#include "pch.h"
#include "include/PowerUpBlock1Object.h"
#include "GameLibrary/GameLibrary.h"
using namespace Library;

TileType PowerUpBlock1Object::GetTileType() {

    return TileType::PowerUp1;
}

void PowerUpBlock1Object::OnHit() {

    isDisappearance = true;
}

void PowerUpBlock1Object::Draw(Renderer& renderer, int x, int y, float offsetY, const Camera& camera) {

    GameLibrary::DrawBlock(renderer,x,y,offsetY,camera,TILE_SIZE,ColorF::Red);
}

File: 99_AI生成ソフトウェア(バイブコーディング)/1_Webアプリ系(VSCode)/3_SideViewGameSample_TypeScript_移植元CPlus(Claude)/SideViewGameSample_移植元CPlus/src/GameLogic/GameObject/PowerUpBlock1Object.ts
import { Renderer } from "../Renderer";
import { Camera } from "../Camera";
import { TileType } from "../TileType";
import { ColorF } from "../ColorF";
import { TileMapObject } from "./TileMapObject";
import { GameLibrary } from "../../GameLibrary/GameLibrary";

/**
 * PowerUpBlock1Object.ts
 * 移植元: include/PowerUpBlock1Object.h / src/PowerUpBlock1Object.cpp
 *
 * パワーアップブロック(赤の疑似立方体)。接触(OnHit)すると
 * isDisappearance_ が true になり、以降マップから消滅する。
 */
export class PowerUpBlock1Object extends TileMapObject {

    public override getTileType(): TileType {
        return TileType.PowerUp1;
    }

    public override onHit(): void {
        this.isDisappearance_ = true;
    }

    public override draw(renderer: Renderer, x: number, y: number, offsetY: number, camera: Camera): void {
        GameLibrary.drawBlock(renderer, x, y, offsetY, camera, TileMapObject.TILE_SIZE, ColorF.Red);
    }
}
